src/main.rs

```diff
@@ -49,15 +49,95 @@ use std::io::ErrorKind;
 use std::os::unix::fs::PermissionsExt;
 use std::os::unix::net::{UnixListener, UnixStream};
 use std::path::Path;
-use std::thread;
+use std::time::Duration;
 
 use anyhow::{Context, Result};
+use crossbeam_channel as channel;
 use slog::{debug, error, o, Drain};
+use threadpool::ThreadPool;
 
 mod ffi;
 mod handlers;
 mod protocol;
 
+fn main() -> Result<()> {
+    const SOCKET_PATH: &str = "/var/run/nscd/socket";
+    const N_WORKERS: usize = 256;
+    const HANDOFF_TIMEOUT: Duration = Duration::from_secs(3);
+
+    ffi::disable_internal_nscd();
+
+    let decorator = slog_term::TermDecorator::new().build();
+    let drain = slog_term::FullFormat::new(decorator).build().fuse();
+    let drain = slog_async::Async::new(drain).build().fuse();
+
+    let logger = slog::Logger::root(drain, slog::o!());
+
+    let (pool, handle) = worker_pool(logger.clone(), N_WORKERS);
```

Comment:
Can we call `handle` `channel` or `tx` or something instead? "handle" is so generic